api/task/ingest.go

```diff
@@ -635,3 +637,52 @@ func getUniqueDatasetName(meta *model.Metadata, storage api.MetadataStorage) (st
 
 	return getUniqueString(meta.Name, datasetNames), nil
 }
+
+func verifyData(pg *postgres.Database, meta *model.Metadata, storage api.MetadataStorage, tableName string) {
```

Comment:
If possible, it may be better to put this function in the postgres storage package. Then the caller would use the data storage interface to call the function. The postgres storage already has the metadata storage as struct property so no need to pass anything else around.